tests/frontier/opcodes/test_calldataload.py

```diff
@@ -0,0 +1,124 @@
+    Transaction,
+)
+from ethereum_test_tools import Opcodes as Op
+
+
+@pytest.mark.parametrize(
+    "calldataload_offset,memory_preset_code,call_args_size,expected_calldataload_memory",
+    [
+        (
+            0x00,
+            Bytecode(Op.MSTORE8(offset=0x0, value=0x25) + Op.MSTORE8(offset=0x1, value=0x60)),
+            0x02,
+            0x2560000000000000000000000000000000000000000000000000000000000000,
+        ),
+        (
+            0x01,
+            Bytecode(
+                Op.MSTORE(
+                    offset=0x0,
+                    value=0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF,
+                )
+                + Op.MSTORE8(offset=0x20, value=0x23),
+            ),
+            0x21,
+            0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF23,
+        ),
+        (
+            0x05,
+            Bytecode(
+                Op.MSTORE(
+                    offset=0x0,
+                    value=0x123456789ABCDEF0000000000000000000000000000000000000000000000000,
+                )
+                + Op.MSTORE8(offset=0x20, value=0x0)
+                + Op.MSTORE8(offset=0x21, value=0x24),
+            ),
+            0x22,
+            0xBCDEF00000000000000000000000000000000000000000000000000024000000,
+        ),
+    ],
+    ids=["two_bytes", "word_n_byte", "34_bytes"],
+)
+@pytest.mark.valid_from("Frontier")
+def test_calldataload(
+    state_test: StateTestFiller,
+    fork: str,
+    calldataload_offset: int,
+    memory_preset_code: Bytecode,
+    call_args_size: int,
+    expected_calldataload_memory: Account,
+    pre: Alloc,
+):
+    """
+    Test `CALLDATACOPY` opcode.
+
+    Based on https://github.com/ethereum/tests/blob/develop/src/GeneralStateTestsFiller/VMTests/vmTests/calldataloadFiller.yml
+    """
+    env = Environment()
+    sender = pre.fund_eoa()
+    post = {}
+
+    # Deploy the contract that will store the calldate
+    sstore_contract = pre.deploy_contract(
+        code=(Op.SSTORE(key=0x0, value=Op.CALLDATALOAD(offset=calldataload_offset)))
+    )
+
+    # Deploy the contract that will forward the calldata to the first contract
+    calldata_contract = pre.deploy_contract(
+        code=(
+            memory_preset_code
+            + Op.CALL(
+                gas=Op.SUB(Op.GAS, 20_000),
+                address=sstore_contract,
+                value=0x0,
+                args_offset=0x0,
+                args_size=call_args_size,
+                ret_offset=0x0,
+                ret_size=0x0,
+            )
+        )
+    )
+
+    validation_contract = pre.deploy_contract(
+        code=(
+            Op.CALL(
+                gas=Op.SUB(Op.GAS, 20_000),
+                address=calldata_contract,
+                value=0x0,
+                args_offset=0x0,
+                args_size=0x0,
+                ret_offset=0x0,
+                ret_size=0x0,
+            )
+        )
+    )
+
+    tx = Transaction(
+        data=b"\x01",
```

Comment:
This data is not actually used right now, so we can remove it.

_But..._ we could write a second test where test `Op.CALLDATALOAD` on the first call stack level:

Right now we are testing on the second call stack level, simply because we are using the opcode on a second contract via `Op.CALL`.

Another test could use this transaction field to test `Op.CALLDATALOAD` using a single contract.